warehouse/oidc/services.py

```diff
@@ -148,10 +148,51 @@ def get_key(self, key_id):
                 tags=[f"provider:{self.provider}", f"key_id:{key_id}"],
             )
             return None
-        return PyJWK(keyset[key_id])
+        return jwt.PyJWK(keyset[key_id])
+
+    def _get_key_for_token(self, token):
+        """
+        Return a JWK suitable for verifying the given JWT.
+
+        The JWT is not verified at this point, and this step happens
+        prior to any verification.
+        """
+        unverified_header = jwt.get_unverified_header(token)
+        return self.get_key(unverified_header["kid"])
 
     def verify(self, token):
```

Comment:
N.B.: It might make sense to rename this to `verify_signature` (or even `verify_signature_only`), to emphasize that this service is only providing a signature verification check, not authN or authZ for a specific action.

Reply:
Agreed